hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -367,10 +367,12 @@ private Pair<Boolean, List<CleanFileInfo>> getFilesToCleanKeepingLatestCommits(S
           }
         }
       }
+      boolean noReplacedFileAfterOrOnEarliestRetainInstant =
+              !fileSystemView.getReplacedFileGroupsAfterOrOn(earliestCommitToRetain.getTimestamp(), partitionPath).findAny().isPresent();
       // if there are no valid file groups
```

Comment:
Then when the partition dir can be removed?

Reply:
Good point, will do